authors/apps/articles/views.py

```diff
@@ -0,0 +1,47 @@
+from django.shortcuts import render
+from rest_framework.generics import ListCreateAPIView, RetrieveUpdateDestroyAPIView
+from rest_framework import status
+from rest_framework.permissions import IsAuthenticated, IsAuthenticatedOrReadOnly
+from .serializers import ArticleSerializer
+from .models import Article
+from authors.apps.authentication.models import User
+from rest_framework.response import Response
+
+
+class ArticleView(ListCreateAPIView):
+    serializer_class = ArticleSerializer
+    permission_classes = (IsAuthenticated,)
+    queryset = Article.objects.all()
+
+    def post(self, request):
+        article = request.data.get("article", {})
+
+        serializer = self.serializer_class(data=article)
+        serializer.is_valid(raise_exception=True)
+        serializer.save(
+            author=User.objects.filter(username=request.user.username).first()
+        )
+        return Response(serializer.data, status=status.HTTP_201_CREATED)
+
+class ArticleRetrieveUpdateDestroy(RetrieveUpdateDestroyAPIView):
+    serializer_class = ArticleSerializer
+    permission_classes = (IsAuthenticated, )
+    queryset = Article.objects.all()
+    lookup_field = 'slug'
+
+    def update(self, request, *args, **kwargs):
+        article = Article.objects.filter(slug=kwargs.get('slug')).first()
+        print(article)
```

Comment:
Hey, @Emmanuelaaron  l suggest that you remove this print statement unless it is doing something that l do not know about, l stand to be correctified

Reply:
@jennizalwango Thanks for the observation. I have resolved the issue